src/sampling/homogeneous.jl

```diff
@@ -12,11 +12,13 @@ struct HomogeneousSampling <: ContinuousSamplingMethod
   size::Int
```

Comment:
What about a new field `weightfun::F` that is initialized with `measure` by default? Or perhaps a new field `weights::W` that is initialized with `nothing` by default and computed inside the method with `measure`?

I think we can avoid introducing an internal function and instead try to add an option to the struct method.